### PR TITLE
feat: Avatar.Group adding max props

### PR DESCRIPTION
**Pull request by wanpan11**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

- close https://github.com/ant-design/ant-design/issues/48950

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     feat: Avatar.Group adding max props      |
| 🇨🇳 Chinese |   Avatar.Group 新增 max 属性    |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
需要废弃掉老的 API。

File: components/avatar/index.zh-CN.md [hunk 1]
```diff
@@ -54,6 +54,7 @@ group:
 
 | 参数 | 说明 | 类型 | 默认值 | 版本 |
 | --- | --- | --- | --- | --- |
+| max | 设置最多显示相关配置 | `{ count?: number; style?: CSSProperties; popover?: PopoverProps }` | - | 5.18.0 |
 | maxCount | 显示的最大头像个数 | number | - |  |
 | maxPopoverPlacement | 多余头像气泡弹出位置 | `top` \| `bottom` | `top` |  |
 | maxPopoverTrigger | 设置多余头像 Popover 的触发方式 | `hover` \| `focus` \| `click` | `hover` | 4.17.0 |
```
Comment (afc163) [inline components/avatar/index.zh-CN.md:60]: 老的 API 移除吧。

Comment (wanpan11) [reply components/avatar/index.zh-CN.md:60]: 好的

File: components/avatar/group.tsx [hunk 2]
```diff
@@ -32,10 +34,19 @@ export interface GroupProps {
   children?: React.ReactNode;
   style?: React.CSSProperties;
   prefixCls?: string;
+  /** @deprecated Please use `max` */
   maxCount?: number;
+  /** @deprecated Please use `max` */
   maxStyle?: React.CSSProperties;
+  /** @deprecated Please use `max` */
```
Comment (afc163) [inline components/avatar/group.tsx:41]:
```suggestion
  /** @deprecated Please use `max={{ style: { ... } }}` */
```

Comment (afc163) [reply components/avatar/group.tsx:41]: 其他属性类似修改。

File: components/avatar/index.zh-CN.md [hunk 3]
```diff
@@ -54,10 +54,7 @@ group:
 
 | 参数 | 说明 | 类型 | 默认值 | 版本 |
 | --- | --- | --- | --- | --- |
-| maxCount | 显示的最大头像个数 | number | - |  |
-| maxPopoverPlacement | 多余头像气泡弹出位置 | `top` \| `bottom` | `top` |  |
-| maxPopoverTrigger | 设置多余头像 Popover 的触发方式 | `hover` \| `focus` \| `click` | `hover` | 4.17.0 |
-| maxStyle | 多余头像样式 | CSSProperties | - |  |
+| max | 设置最多显示相关配置 | `{ count?: number; style?: CSSProperties; popover?: PopoverProps }` | - | 5.18.0 |
```
Comment (afc163) [inline components/avatar/index.zh-CN.md:57]: 这里可以写一下 5.18.0 之前怎么写，然后留一个 github 代码地址链接

**Comment by wanpan11** [comment]
改好啦

File: components/avatar/index.en-US.md [hunk 4]
```diff
@@ -49,10 +49,7 @@ Common props ref：[Common props](/docs/react/common-props)
 
 | Property | Description | Type | Default | Version |
 | --- | --- | --- | --- | --- |
-| maxCount | Max avatars to show | number | - |  |
-| maxPopoverPlacement | The placement of excess avatar Popover | `top` \| `bottom` | `top` |  |
-| maxPopoverTrigger | Set the trigger of excess avatar Popover | `hover` \| `focus` \| `click` | `hover` | 4.17.0 |
-| maxStyle | The style of excess avatar style | CSSProperties | - |  |
+| max | Set maximum display related configurations, Before `5.18.0` you can use [parameters](https://github.com/ant-design/ant-design/blob/master/components/avatar/group.tsx#L37-L44) | `{ count?: number; style?: CSSProperties; popover?: PopoverProps }` | - | 5.18.0 |
```
Comment (afc163) [inline components/avatar/index.en-US.md:52]: https://github.com/ant-design/ant-design/blob/9d134859becbdae5b9ce276f6d9af4264691d81f/components/avatar/group.tsx#L37-L44

Comment (afc163) [inline components/avatar/index.en-US.md:52]:
```suggestion
| max | Set maximum display related configurations, Before `5.18.0` you can use [parameters](https://github.com/ant-design/ant-design/blob/9d134859becbdae5b9ce276f6d9af4264691d81f/components/avatar/group.tsx#L37-L44) | `{ count?: number; style?: CSSProperties; popover?: PopoverProps }` | - | 5.18.0 |
```

**Comment by wanpan11** [comment]
好啦

File: components/avatar/group.tsx [hunk 5]
```diff
@@ -81,22 +102,29 @@ const Group: React.FC<GroupProps> = (props) => {
...
+
+    const mergeStyle = max?.style || maxStyle;
+    const mergeProps = {
+      content: childrenHidden,
+      trigger: maxPopoverTrigger,
+      placement: maxPopoverPlacement,
+      ...max?.popover,
+      overlayClassName: max?.popover?.overlayClassName
+        ? `${groupPrefixCls}-popover ${max?.popover?.overlayClassName}`
+        : `${groupPrefixCls}-popover`,
```
Comment (afc163) [inline components/avatar/group.tsx:119]: 用 classNames

File: components/avatar/group.tsx [hunk 6]
```diff
@@ -58,8 +69,18 @@ const Group: React.FC<GroupProps> = (props) => {
...
+    max,
   } = props;
 
+  if (process.env.NODE_ENV !== 'production') {
+    const warning = devUseWarning('Avatar.Group');
+    warning.deprecated(
+      !maxCount || !maxStyle || !maxPopoverPlacement || !maxPopoverTrigger,
+      'maxCount maxStyle maxPopoverPlacement maxPopoverTrigger',
+      'max',
+    );
```
Comment (afc163) [inline components/avatar/group.tsx:81]: 每个属性分开说明，堆在一起不好找。

**Review by afc163 — DISMISSED** [review]

**Comment by wanpan11** [comment]
默认值没去掉 难怪 ci 挂了 🤧

**Review by afc163 — APPROVED** [review]

**Pull request merged**